docs/standard/analyzers/platform-compat-analyzer.md

```diff
@@ -0,0 +1,363 @@
+
+  // an API was unsupported on windows until version 8.0.
+  // The API is considered supported everywhere else without constraints.
+  [UnsupportedOSPlatform("windows")]
+  [SupportedOSPlatform("windows8.0")]
+  public void StartedWindowsSupportFromVersion8() { }
+
+  // an API was unsupported on windows until version8.0.
+  // Then the API is removed (unsupported) from version 10.0.19041.0.
+  // The API is considered supported everywhere else without constraints.
+  [UnsupportedOSPlatform("windows")]
+  [SupportedOSPlatform("windows8.0")]
+  [UnsupportedOSPlatform("windows10.0.19041.0")]
+  public void StartedWindowsSupportFrom8UnsupportedFrom10() { }
+
+  public void Caller2()
+  {
+      DoesNotWorkOnAndroid(); // warns 'DoesNotWorkOnAndroid' is unsupported on 'android'
+
+      // warns:'StartedWindowsSupportFromVersion8' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFromVersion8' is supported on 'windows' 8.0 and later
+      StartedWindowsSupportFromVersion8();
+
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is supported on 'windows' 8.0 and later
+      // even there were 3 diagnostics found analyzer warn only for the first 2
+      StartedWindowsSupportFrom8UnsupportedFrom10();
+  }
+  ```
+
+### How to handle the warnings reported?
+
+The recommend way to deal with these diagnostics is by making sure you only call these APIs when running on the appropriate platforms. You have five options on how you can address these warnings, choose which ever appropriate for your situation.
+
+1. **Guard the call**. You can either achieve this by excluding the code at build time using `#if` and multi-targeting or by conditionally calling the code at runtime. Check whether you’re running on desired `Platform` by using any of platform check methods, like `OperatingSystem.Is<Platform>()` or `OperatingSystem.Is<Platform>VersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)`.
+
+2. **Mark the call site as platform-specific**. You can also choose to mark your own APIs as being platform-specific, thus effectively just forwarding the requirements to your callers. You can mark the containing method or type or the entire assembly with same attributes as of the referenced platform-dependent call. [Examples](#call-site-marked-as-platform-specific-for-platform-specific-api-calls-in-above-examples)
+
+3. **Assert the call site with platform check**. If you don't want the overhead of an additional if statement at run time, use <xref:System.Diagnostics.Debug.Assert(System.Boolean)?displayProperty=nameWithType>. [Example](#assert-the-call-site-with-platform-check)
+
+4. **Delete the code**. Generally not what you want because it means you lose fidelity when your code is used by Windows users, but for cases where a cross-platform alternative exists, you’re likely better off using that over platform-specific APIs.
```

Comment:
```suggestion
- **Delete the code**. Generally not what you want because it means you lose fidelity when your code is used by Windows users. For cases where a cross-platform alternative exists, you’re likely better off using that over platform-specific APIs.
```